File: serverlist.py
#!/bin/env python
import socket
import random

def get_radiobrowser_base_urls():
    """
    Get all base urls of all currently available radiobrowser servers

    Returns: 
    list: a list of strings

    """
    hosts = []
    # get all hosts from DNS
    ips = socket.getaddrinfo('all.api.radio-browser.info',
                             80, 0, 0, socket.IPPROTO_TCP)
    for ip_tupple in ips:
        ip = ip_tupple[4][0]

        # do a reverse lookup on every one of the ips to have a nice name for it
        host_addr = socket.gethostbyaddr(ip)
        # add the name to a list if not already in there
        if host_addr[0] not in hosts:
            hosts.append(host_addr[0])

    # sort list of names
    hosts.sort()
    # add "https://" in front to make it an url
    return list(map(lambda x: "https://" + x, hosts))

def get_radiobrowser_base_url_random():
	"""
    Get a random available base url

    Returns: 
    str: a random available base url

    """
	hosts = get_radiobrowser_base_urls()
	return random.choice(hosts)

# print list of names
print("All available urls")
print("------------------")
for host in get_radiobrowser_base_urls():
    print(host)
print("")

# print random
print("Random url")
print("------------------")
print(get_radiobrowser_base_url_random())

File: serverlist.js
#!/bin/env node

const dns = require('dns');
const util = require('util');
const resolve4 = util.promisify(dns.resolve4);
const reverse = util.promisify(dns.reverse);

/**
 * Get a list of base urls of all available radio-browser servers
 * Returns: array of strings - base urls of radio-browser servers
 */
function get_radiobrowser_base_urls() {
    return resolve4("all.api.radio-browser.info").then(hosts => {
        let jobs = hosts.map(host => reverse(host));
        return Promise.all(jobs);
    }).then(hosts => {
        hosts.sort();
        return hosts.map(host_arr => "https://" + host_arr[0]);
    });
}

/**
 * Get a random available radio-browser server.
 * Returns: string - base url for radio-browser api
 */
function get_radiobrowser_base_url_random() {
    return get_radiobrowser_base_urls().then(hosts => {
        var item = hosts[Math.floor(Math.random() * hosts.length)];
        return item;
    });
}

get_radiobrowser_base_urls().then(hosts => {
    console.log("All available urls")
    console.log("------------------")
    for (let host of hosts) {
        console.log(host);
    }
    console.log();

    return get_radiobrowser_base_url_random();
}).then(random_host => {
    console.log("Random base url")
    console.log("------------------")
    console.log(random_host);
});